src/collectors/AbstractCollector.php

```diff
@@ -118,41 +138,48 @@ public function getMinTime($last = null) : DateTimeImmutable
         return $this->minTime;
     }
 
-    protected function findMinTime($last) : DateTimeImmutable
+    /**
+     * Calculate time
+     *
+     * @param string $last
+     * @return DateTimeImmutable
+     */
+    protected function findMinTime(string $last) : DateTimeImmutable
     {
         $laststamp = strtotime($last);
         $minTime = $this->getMinTime();
-        $minTimeStamp = ($minTime instanceof DateTimeImmutable)
-            ? $minTime->getTimestamp()
-            : (is_int($minTime) ? $minTime : 0);
 
-        if ($laststamp && $laststamp < $minTimeStamp) {
+        if ($laststamp && $laststamp < $minTime->getTimestamp()) {
             return DateTimeImmutable($last);
         }
 
         return $this->getMinTime();
     }
 
-    protected function buildTime($time, $default) : DateTimeImmutable
+    /**
+     * Create timestamp
+     *
+     * @param string $time
+     * @param string $default
+     * @return DateTimeImmutable
+     */
+    protected function buildTime(?string $time = '', string $default) : DateTimeImmutable
     {
         if (isset($time)) {
```

Comment:
вместо всего ифа можно просто
```suggestion
        $time = $time ?? $default
```

Reply:
не так